Restructure the readme to make it easier to find out how to track different items (#264)
Financial-Times/o-tracking: README.md

        - `membership` - All membership pages
        - `page` - anything else, not above.
    */
    asset_type: "story"
  }
};
oTracking.page(pageConfig);
```

Call the `oTracking.click.init` method to track click events. Pass the category you would like the click events to have as an argument:

- If the element being clicked is a link which goes to a page on the same domain as the current page, o-tracking will put the tracking data into a queue to send to Spoor at a later time.
- Add the attribute `data-o-tracking-skip-queue` to the element to send the data to Spoor immediately.
- O-tracking click events will also track the path from the root of the DOM tree to the element which was clicked. This is recorded in a property called `domPathTokens`.
- If the clicked element has the `data-trackable` attribute set, sibling elements will also be included within the `domPathTokens` property.
```js
// Tracking a click
const category = 'element';
oTracking.click.init(category);
```

To track when an element has come into view of the user, add the attribute `data-o-tracking-view` to the element in the page and then call the `oTracking.view.init` method:

> * View events are fired for elements with the `data-o-tracking-view` attribute by default, unless `o-tracking`'s `selector` option is configured. Like click events, view events populate a `domPathTokens` property. To collect data for events, set the `category` option, or provide a callback[`getContextData`]

Author: ```suggestion
- By default, elements with the `data-o-tracking-view` attribute are tracked.
- To track different elements, set the `selector` option property to a CSS selector.
- Like click events, view events will also track the path from the root of the DOM tree to the element which triggered the tracking event into a property called `domPathTokens`.
- To categorise the view events, set the `category` option property.
- To collect extra data to send with the tracking event, add a function named `getContextData` to the options. The function receives as it's single argument the element which triggered the tracking event and needs to return an object with any of these optional properties set:
	- `componentContentId`
	- `type`
	- `subtype`
	- `component`
```

@notlee How about this? ☝️

Reviewer: great :)

	<div style="background: url('https://spoor-api.ft.com/px.gif?data=YOUR_URL_ENCODED_JSON_DATA_HERE');"></div>
</div>
```

For example, if you have the following data you want to track in Spoor:

```json
{
  "category": "page",
  "action": "view",
  "system": {
    "apiKey": "qUb9maKfKbtpRsdp0p2J7uWxRPGJEP",
    "source": "o-tracking",
    "version": "1.0.0"
  },
  "context": {
    "product": "ft.com",
    "content": {
      "asset_type": "page"
    }
  }
}
```
Here is the corresponding tracking pixel setup:
```html
<div class="o-tracking o--if-no-js" data-o-component="o-tracking">
    <div style="background: url('https://spoor-api.ft.com/px.gif?data=%7B%22category%22:%22page%22,%20%22action%22:%22view%22,%20%22system%22:%7B%22apiKey%22:%22qUb9maKfKbtpRsdp0p2J7uWxRPGJEP%22,%22source%22:%22o-tracking%22,%22version%22:%221.0.0%22%7D,%22context%22:%7B%22product%22:%22ft.com%22,%22content%22:%7B%22asset_type%22:%22page%22%7D%7D%7D');"></div>
</div>
```

### Tracking with JavaScript

To manually instantiate `o-tracking`, import the component and call the `init` method with configuration which is specific to your product:

> If o-tracking is running on a page which has an ftsession cookie, it will automatically add that information to the data it is tracking.

Reviewer: This line feels a little out of place between the `init` instructions above and the code example below. I'm not sure if it belongs as part of the previous `init` paragraph or somewhere else?

Author: Yeah, I think you are right, it needs to be separate from `init`.

> * Automatically pickup ftsession from cookies for you.

Reviewer: ```suggestion
```

const pageConfig = {
  content: {
      /*
        Asset type is meant to describe the main purpose of the page
        The value can be one of these:
        - `story` - A story or article
        - `blog` - A blog post
        - `front` - A home page or front page
        - `ad` - An advert.
        - `image` - An image
        - `interactive` - An interactive graphic
        - `report` - A special report
        - `search` - A search results page
        - `section` - A section or listing page
        - `topic` -  A topic landing page
        - `video` - A video page
        - `login` - Any login/sign-in page
        - `stream` - A stream page
        - `funnel` - Any funnel page
        - `epaper` - all epaper pages
        - `rankings` - A rankings page for schools and courses (i.e. on rankings.ft.com). Not the section's hub page
        - `markets` - Any market, bond, commoditity, stock, currency 'tearsheet' (usually has "/tearsheet/" in URL)
        - `myft` - All MyFT pages
        - `account` - All account pages
        - `membership` - All membership pages
        - `page` - anything else, not above.
    */
    asset_type: "story"
  }
};
oTracking.page(pageConfig);
```

> To track click events you call the `oTracking.click.init` method and pass it the category you would like the events to have:

Reviewer: >To track click events you call the `oTracking.click.init` method and pass it the category you would like the events to have:

Could be written more directly:

>Call the `oTracking.click.init` method to track click events. Pass the category you would like the click events to have as an argument:

At first I thought this is an example of [passive voice which we try to avoid](https://origami.ft.com/docs/principles/tone-and-language/#use-the-active-voice), but now I'm not sure.

- If the element being clicked is a link which goes to a page on the same domain as the current page, o-tracking will put the tracking data into a queue to send to Spoor at a later time.

> - If you want to send the data to Spoor immediately, you can add the attribute `data-o-tracking-skip-queue` onto the element.

Reviewer: This is another example that could be written more directly. What do you think? I wish I knew where I picked up this habit. I'm not sure if it's one of our documentation principles. I think it's [passive voice](https://www.grammarly.com/blog/active-vs-passive-voice/), right?

>If you want to send the data to Spoor immediately, you can add the attribute `data-o-tracking-skip-queue` onto the element.

could be

>Add the attribute `data-o-tracking-skip-queue` to the element to send the data to Spoor immediately.

- O-tracking click events will also track the path from the root of the DOM tree to the element which was clicked. This is recorded in a property called `domPathTokens`.
- If the clicked element has the `data-trackable` attribute set, sibling elements will also be included within the `domPathTokens` property.
```js
// Tracking a click
const category = 'element';
oTracking.click.init(category);
```

To track when an element has come into view of the user, add the attribute `data-o-tracking-view` to the element in the page and then call the `oTracking.view.init` method:

> * View events are fired for elements with the `data-o-tracking-view` attribute by default, unless `o-tracking`'s `selector` option is configured. Like click events, view events populate a `domPathTokens` property. To collect data for events, set the `category` option, or provide a callback[`getContextData`]

Reviewer: maybe this could be split into multiple bullet points?

_Note:_ This feature requires `window.IntersectionObserver` in order to track the events
_Note:_ `getContextData` should be a function which returns `{Object}`. It accepts the viewed element as an argument

```js
const opts = {
    category: 'audio', // default: 'component'
    selector: '.o-teaser__audio', // default: '[data-o-tracking-view]'
    getContextData: (el) => {  // default: null
        return {
            componentContentId: el.getAttribute('data-id'),
            type: 'audio',
            subtype: 'podcast',
            component: 'teaser'
        };
    }
};

oTracking.view.init(opts);
```

> To track a custom event you call the `oTracking.event` method:

Reviewer: ```suggestion
To track a custom event call the `oTracking.event` method:
```

```js
const eventConfig = {
    "category": "video",
    "action": "play",
};
oTracking.event(eventConfig);
```

#### Events

> You can configure o-tracking to listen for `oTracking.page` and `oTracking.event` events instead of using the `page` and `event` methods.

Reviewer: This sentenced confused me at first. It might be worth being a little more verbose:

```suggestion
Instead of calling the `page` and `event` methods `o-tracking` can be configured to fire events based on the custom DOM Events `oTracking.page` and `oTracking.event`.
```

Author: This is much better :-)

##### oTracking.page

> To make o-tracking listen for `oTracking.page` events, you call `oTracking.page.init()`:

Reviewer: ```suggestion
Call the `oTracking.page.init()` method to listen for `oTracking.page` events :
```

```js
// Tell o-tracking to listen for `oTracking.page` events
oTracking.page.init();

// Now we can send page events and o-tracking will track the data within them.
const pageData = { content: { uuid: 'abc-123', barrier: 'PREMIUM' };
document.body.dispatchEvent(new CustomEvent('oTracking.page', { detail: pageData}, bubbles: true}));
```

##### oTracking.event

To make o-tracking listen for `oTracking.event` events, you call `oTracking.event.init()`:
```js
// Tell o-tracking to listen for `oTracking.event` events
oTracking.event.init();

// Now we can send custom events and o-tracking will track the data within them.
const customData = { category: 'video', action: 'play', id: '512346789', pos: '10' };
document.body.dispatchEvent(new CustomEvent('oTracking.event', { detail: customData}, bubbles: true}));
```

### Example implementations

- [ft.com](docs/ftcom_example.md)

> - [membership](docs/membership_example.md)

Reviewer: The line after this "Events are decorated with config values [...]" looks like its part of the "Example implementations" section. Needs a new heading?

Author: I'll place that content in the event.md file 👍
